dtfs-central-api/src/v1/controllers/tfm/deal/tfm-submit-deal.controller.js

```diff
@@ -90,11 +90,55 @@ const createFacilitiesSnapshot = async (deal) => {
   return { status: 400, message: 'Invalid Deal Id' };
 };
 
+/**
+ * creates amendments object in tfm-facilities
+ * object blank history array
+ */
+const createFacilitiesAmendments = async (deal) => {
+  if (ObjectId.isValid(deal._id)) {
+    const { dealType, _id: dealId } = deal;
+
+    let dealFacilities = [];
+    if (dealType === CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS) {
+      dealFacilities = await findAllFacilitiesByDealId(dealId);
+    }
+
+    if (dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF) {
+      dealFacilities = await findAllGefFacilitiesByDealId(dealId);
+    }
+
+    const collection = await db.getCollection('tfm-facilities');
+
+    // blank amendments object
+    const amendmentsObject = {
+      history: [],
+    };
+
+    if (dealFacilities) {
+      const updatedAmendments = Promise.all(
+        dealFacilities.map(async (facility) =>
```

Comment:
I thought we need to add the amendment for the existing facility. 
What happens if a deal has 2 or facilities? Isn't this going to "append" the `amendments` object to them?

Reply:
this adds for all the facilities and adds the amendments: {history: []} to all the tfm-facilities as part of a deal

Reply:
removed